### PR TITLE
Fix/deferred payment methods

### LINKED ISSUE #1837: Payment Methods inconsistencies when using PaymentSheet in deferred mode
When using PaymentSheet in 'Deferred Mode' the allowed payment methods can't come from intent, because the intent will be generated only after the user has choosen a payment method. Payment methods that will be shown should are those selected in Stripe Settings. From code we should be able to further filtering allowed payment methods by giving an array when istantiating `IntentConfiguration`.

With current code we have two problems:

1. In iOS, payment methods like iDeal and PayPayl won't be shown
2. In Android all configured payment methods are shown, regardless of wich values we put in `paymentMethodTypes` array in `intentConfiguration`

**To Reproduce**
1. Enable  PayPal, an other payment types in your Stripe account
2. Add `paymentMethodTypes: ['card','paypal'],` in `IntentConfiguration` at 'initPaymentSheet' method of payment_sheet_deffered_screen.dart file
3. Run the example App
4. Launch 'Single Step (deferred payment)' demo from 'Payment Sheet' group
7. In iOS: Observe that only 'Card' Payment is shown
8. In Android: Observe that all configured payment methods are shown, even if not present in `paymentMethodTypes` array.

Solution:

1. iOS: A return Url should be configured when configuring the payment sheet
2. Android: JSONArray is missing from possible tipes in ReadableMap.java from stripe_android package

A PR will be submitted to address this issue.

### PR DESCRIPTION
**Pull request by pgiacomo69**
Fixes #1837

### CODE REVIEW

**Comment by remonh87** [comment]
thank you for your contribution however we do not change the underlying stripe sdk. If you want this behavior to change you should create an pr for https://github.com/stripe/stripe-react-native .

**Comment by jonasbark** [comment]
We will however merge the Android part

**Comment by pgiacomo69** [comment]
Hi @remonh87 , @jonasbark do you think that the new commit can solve the issue? 
Thank you 👍

File: packages/stripe_ios/ios/Classes/Stripe Sdk/StripeSdk+PaymentSheet.swift [hunk 1]
```diff
@@ -224,9 +224,11 @@ extension StripeSdk {
             mode = PaymentSheet.IntentConfiguration.Mode.payment(
                 amount: amount,
                 currency: modeParams["currencyCode"] as? String ?? "",
-                setupFutureUsage: modeParams["setupFutureUsage"] != nil
-                    ? (modeParams["setupFutureUsage"] as? String == "OffSession" ? .offSession : .onSession)
-                    : nil,
+                setupFutureUsage: { return switch modeParams["setupFutureUsage"] as? String {
```
Comment (remonh87) [inline packages/stripe_ios/ios/Classes/Stripe Sdk/StripeSdk+PaymentSheet.swift:227]: can you remove this change and let's go for a fix in dart or we need to make a change to stripe ios

**Review by remonh87 — CHANGES_REQUESTED** [review]

File: packages/stripe_ios/ios/Classes/Stripe Sdk/StripeSdk+PaymentSheet.swift (hunk 1 above)
Comment (pgiacomo69) [reply packages/stripe_ios/ios/Classes/Stripe Sdk/StripeSdk+PaymentSheet.swift:227]: I thought to have removed it in last commit, while fixing it in dart, let me check when I return home

Comment (pgiacomo69) [reply packages/stripe_ios/ios/Classes/Stripe Sdk/StripeSdk+PaymentSheet.swift:227]: But, a moment, this seems the rollback, compare it against  main branch

Comment (pgiacomo69) [reply packages/stripe_ios/ios/Classes/Stripe Sdk/StripeSdk+PaymentSheet.swift]: Ok, check it now again, please

**Review by remonh87 — APPROVED** [review]

**Pull request merged**